File: Core_PetBoarding_Backend/PetBoarding_Api/Dto/Users/LoginResponseDto.cs
namespace PetBoarding_Api.Dto.Users
{
    public record LoginResponseDto
    {
        public bool Success { get; init; }
        public string Message { get; init; } = string.Empty;
        public string? Token { get; init; }
        public UserDto? User { get; init; }
    }

    public record UserDto
    {
        public Guid Id { get; init; }
        public string Email { get; init; } = string.Empty;
        public string FirstName { get; init; } = string.Empty;
        public string LastName { get; init; } = string.Empty;
        public string PhoneNumber { get; init; } = string.Empty;
        public bool EmailConfirmed { get; init; }
        public bool PhoneNumberConfirmed { get; init; }
        public string ProfileType { get; init; } = string.Empty;
        public string Status { get; init; } = string.Empty;
        public DateTime CreatedAt { get; init; }
        public DateTime UpdatedAt { get; init; }
    }
}

File: ng_PetBoarding_app/src/app/shared/contracts/auth/login-response.dto.ts
export interface LoginResponseDto {
  success: boolean;
  message?: string;
  token?: string;
  refreshToken?: string;
  user?: UserDto;
}

export interface TokenRefreshResponseDto {
  success: boolean;
  message?: string;
  token?: string;
}

export interface AddressDto {
  id?: string;
  streetNumber: string;
  streetName: string;
  city: string;
  postalCode: string;
  country: string;
  complement?: string;
}

export interface UserDto {
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phoneNumber: string;
  profileType: string;
  status: string;
  address?: {
    streetNumber: string;
    streetName: string;
    city: string;
    postalCode: string;
    country: string;
    complement?: string;
  };
  createdAt: Date;
  updatedAt: Date;
}
